pallets/automation-time/src/lib.rs

```diff
@@ -970,54 +1007,62 @@ pub mod pallet {
 		///
 		/// Returns a vec with the tasks that were not run and the remaining weight.
 		pub fn run_tasks(
-			mut task_ids: Vec<T::Hash>,
+			mut account_task_ids: Vec<AccountTaskId<T>>,
 			mut weight_left: Weight,
-		) -> (Vec<T::Hash>, Weight) {
+		) -> (Vec<AccountTaskId<T>>, Weight) {
 			let mut consumed_task_index: usize = 0;
-			for task_id in task_ids.iter() {
+			for account_task_id in account_task_ids.iter() {
```

Comment:
This is used in a lot of other places. I am fine removing the `account_task_id` type all together, but I think it needs to be consistent. This makes more sense to me, but if others find it confusing I am happy to remove the type and always spell it out

Reply:
I appreciated your use of account_task_id in the other places. I thought having a named type to spell it out was really good. Just right here, I think you'll benefit from destructuring the tuple so that in the rest of this method you can explicitly refer to account_id and task_id rather than account_task_id.0 and account_task_id.1

Reply:
Not sure if this will clear up confusion, but this the code I was playing with to make sure that my suggestion was legal rust:
```
type T = (String, u64);
fn main() {
    let x: T = (String::from("a"), 1);
    let y: T = (String::from("b"), 2);
    let v = vec![x, y];
    for (s, n) in v.iter() {
        println!("{s:} {n:}");
    }
}
```

Reply:
I just worry that if all places where we need to refer to different parts of the tuple we destruct it and all others we use the type that it could be confusing on where that type is actually relevant. I can make the change though

Reply:
In my mind that's what the compiler is for. If we define our code with rigorous types then we can default to writing very readable code and the compiler can catch where the developer is misunderstanding the underlying type.